src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Startup.cs

```
@@ -156,13 +156,18 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
 
 #endif
             app.UseStaticFiles();
+
+            app.UseEndpointRouting(routes =>
+            {
+                routes.MapApplication();
+            });
+
             app.UseCookiePolicy();
 
 #if (OrganizationalAuth || IndividualAuth)
             app.UseAuthentication();
-
 #endif
-            app.UseMvc();
+            app.UseAuthorization();
```

Should this be in the block with `UseAuthentication()`? What would this do if you don't have authn?

---

Imagine you have an MVC app without AuthN. You add `[Authorize]` to an action. You get an error from MVC saying that AuthN is not setup.

Now add a non-MVC endpoint, e.g. a `MapGet`. You add `RequiresAuthorization` to it. Should it throw an error saying that AuthN is not setup or should it allow requests to access the endpoint with AuthZ metadata silently?

Crazy idea: I wonder if authz should be baked into Http.Abstractions? i.e. it is always checked before executing the endpoint.